helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -257,6 +258,24 @@ public void updateExternalViews(List<ExternalView> externalViews) {
     }
   }
 
+  /**
+   * Update the cached customized view map
+   * @param customizedViews
+   */
+  public void updateCustomizedViews(String customizedStateType,
+      List<CustomizedView> customizedViews) {
+    for (CustomizedView cv : customizedViews) {
+      if (_customizedViewCacheMap.containsKey(customizedStateType)) {
```

Comment:
You can move this condition out of the loop by checking and adding key if not exists.
Then in the loop, you only need to set property.